src/auth/auth.service.ts

```diff
@@ -19,21 +19,16 @@ export class AuthService {
   ) {}
 
   async register(createUserDto: CreateUserDto): Promise<SanitizedUser> {
-    if (!await this.usersService.validateEmail(createUserDto.email)) {
-      throw new HttpException("Registration error - this email is already used", HttpStatus.BAD_REQUEST);
-    }
-
-    if (!await this.usersService.validateUsername(createUserDto.username)) {
-      throw new HttpException("Registration error - this username is already used", HttpStatus.BAD_REQUEST);
+    try {
+      return await this.usersService.create(createUserDto);
     }
-
-    let user = await this.usersService.create(createUserDto);
-    
-    if (!await this.mailService.sendUserConfirmation(user.email, user.emailNonce)) {
-      throw new HttpException("Email could not be sent", HttpStatus.INTERNAL_SERVER_ERROR);
+    catch (err) {
+      if (err.code === 'P2002' && err.meta?.target?.length > 0) {
```

Comment:
Maybe leave a comment explaining what this means? I'm guessing this is to handle violation of unique keys, but I'm not sure

Reply:
Yes, this code should throw exception if a username or an email is not unique and give user an explanation. I will add a comment to make it clearer.